AmplifyPlugins/API/AWSAPICategoryPlugin/Configuration/AWSAPICategoryPluginConfiguration.swift

```diff
@@ -55,4 +148,28 @@ public struct AWSAPICategoryPluginConfiguration {
 
         return endpoints
     }
+
+    /// Given a dictionary of EndpointConfig indexed by API endpoint name,
+    /// builds a dictionary of EndpointInterceptorsConfig.
+    /// - Parameters:
+    ///   - endpoints: dictionary of EndpointConfig
+    ///   - apiAuthProviderFactory: apiAuthProviderFactory
+    ///   - authService: authService
+    /// - Throws:
+    /// - Returns: dictionary of EndpointInterceptorsConfig indexed by API endpoint name
+    private static func interceptorsForEndpoints(_ endpoints: [APIEndpointName: EndpointConfig],
```

Comment:
low: these are getting confusing to me because this static `interceptorsForEndpoints` is building them while the instance `interceptorsForEndpoint` are used to retrieve it. is there a better naming for these methods? `makeInterceptors(for endpoints:apiAuthProviderFactory:autService:)` and keep the getters `interceptorsForEndpoint`?